### PR TITLE
Add button to opt out of trigger

### PR DESCRIPTION
**Pull request by Teccii**
Makes it so users can opt out of the trigger per-guild. I made a new model for the trigger cuz I got confused by pre-existing ones.

image

### CODE REVIEW

**Comment by Teccii** [comment]
I forgot to actually make it so it opts you out

File: src/models/Trigger.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Model, Schema, model } from "mongoose";
+
+import { ITrigger } from "types/mongodb";
+
+const TriggerSchema = new Schema({
```
Comment (3vilN355) [inline src/models/Trigger.ts:5]: Triggers should probably include the name of the trigger to separate them from the other ones
(so you don't opt out of all triggers all at once)

File: src/events/messageCreate.ts [hunk 2]
```diff
@@ -204,6 +209,15 @@ export default async (client: CustomClient, message: Message): Promise<void> =>
                   "You can find more information about it [here](https://discord.com/channels/474583323340046337/474583324396879884/1067218193912963172)"
               ),
           ],
+          components: [
+            new ActionRowBuilder<ButtonBuilder>()
+              .addComponents(
+                new ButtonBuilder()
+                  .setCustomId(triggerId)
```
Comment (3vilN355) [inline src/events/messageCreate.ts:216]: `triggerId` should probably be related to which trigger it is, maybe something like a `trigger-${triggerName}`

**Review by 3vilN355 — COMMENTED** [review]
Couple feedbacks here, another thought that arises is;
If we're going to have a model for a trigger, we might aswell just store the triggers themselves in the db with the name and keywords array?

Then it's easier to update on the fly down the line if we want to make commands to add/enable/disable/remove em.

**Comment by 3vilN355** [comment]
As for adding them to settings, it's just about adding something like how the chains are added.
You'd find them on the settings object which it always has a reference to in the context where they matter.
Something like a
```ts
    triggers: [{
        id: String,
        keywords: [[String]],
        cooldown: Number
    }],
```
Not sure if `[[String]]` is supported in mongo tho

**Pull request closed without merging**